filer/templates/admin/filer/folder/directory_listing.html

```diff
@@ -147,10 +147,20 @@
                             {% trans "New Folder" %}
                             </a>
                         {% endif %}
+
                         {% if permissions.has_add_children_permission and not folder.is_root %}
                             <a href="#" id="id_upload_button" title="{% trans 'Upload Files' %}" class="navigator-button navigator-button-upload js-upload-button" data-url="{% url 'admin:filer-ajax_upload' folder_id=folder.id %}" data-max-uploader-connections="{{ uploader_connections }}">
                                 {% trans "Upload Files" %}
                             </a>
+                        {% elif folder.name == "unfiled files" %}
+                            <a href="#" id="id_upload_button" title="{% trans 'Upload Files' %}" class="navigator-button navigator-button-upload js-upload-button" data-url="{% url 'admin:filer-ajax_upload' %}" data-max-uploader-connections="{{ uploader_connections }}">
+                                {% trans "Upload Files" %}
+                            </a>
+                        {% endif %}
+                        {% if folder.is_root and folder.name != "unfiled files" %}
+                            <a href="#" title="{% trans 'You can&#39;t upload to the root' %}" class="navigator-button navigator-button-upload" disabled>
```

Comment:
maybe we should do `{{ _('You can&#39;t upload to the root')|espacejs }}` just to be on the safe side